### PR TITLE
Update saitypes.h

### PR DESCRIPTION
**Pull request by trignesh**
Doxygen comment update

### CODE REVIEW

File: inc/saitypes.h [hunk 1]
```diff
@@ -258,36 +252,44 @@ typedef struct _sai_s8_list_t
...
+ *
+ * String should be null terminated and count should include '\0'.
+ */
 typedef struct _sai_u32_list_t
 {
     uint32_t count;
     uint32_t *list;
 } sai_u32_list_t;
 
-typedef struct _sai_s32_list_t
```
Comment (digesh) [inline inc/saitypes.h]: Looks like s32 list is removed by mistake. Please undo

File: inc/saitypes.h [hunk 2]
```diff
@@ -258,36 +252,44 @@ typedef struct _sai_s8_list_t
...
-{
-    uint32_t count;
-    int32_t *list;
-} sai_s32_list_t;
-
-typedef struct _sai_u32_range_t
```
Comment (digesh) [inline inc/saitypes.h]: seems removed wrongly. need to undo

File: inc/saitypes.h [hunk 3]
```diff
@@ -479,16 +498,22 @@ typedef struct _sai_qos_map_params_t
 
 } sai_qos_map_params_t;
 
+/**
+* @brief Defines Qos map data structure contain key value pair.
+*/
 typedef struct _sai_qos_map_t
 {
-    /** Input parameters to match */
+    /** Input map parameters */
```
Comment (digesh) [inline inc/saitypes.h]: change above comment to : "Input qos map parameters match"

File: inc/saitypes.h [hunk 4]
```diff
@@ -258,36 +252,44 @@ typedef struct _sai_s8_list_t
...
+ * String should be null terminated and count should include '\0'.
+ */
 typedef struct _sai_u16_list_t
 {
     uint32_t count;
     uint16_t *list;
 } sai_u16_list_t;
 
+/**
+ * @brief Defines a u16 list or string
```
Comment (digesh) [inline inc/saitypes.h]: Its s16 list NOT u16 list. change it to s16 list in doxygen comment

File: inc/saitypes.h [hunk 5]
```diff
@@ -241,12 +241,6 @@ typedef enum _sai_object_type_t
     SAI_OBJECT_TYPE_MAX                      = 70,
 } sai_object_type_t;
 
-typedef struct _sai_u8_list_t
```
Comment (digesh) [inline inc/saitypes.h]: Shouldn't remove u8 list. Please undo

**Comment by kcudnik** [comment]
Please fix compilation issues

File: inc/saitypes.h [hunk 6]
```diff
@@ -258,36 +263,66 @@ typedef struct _sai_s8_list_t
...
 typedef struct _sai_s16_list_t
 {
     uint32_t count;
     int16_t *list;
 } sai_s16_list_t;
 
+/**
+ * @brief Defines a u32 list or string
+ *
+ * String should be null terminated and count should include '\0'.
```
Comment (kcudnik) [inline inc/saitypes.h]: why other lists except int8 and uin8 are considered strings ?

File: inc/saitypes.h [hunk 7]
```diff
@@ -324,13 +365,20 @@ typedef struct _sai_ip_address_t
...
 
+/**
+ * @brief Defines a IP prefix data structure
+ * contain address and mask
+ */
 typedef struct _sai_ip_prefix_t
 {
+    /** IP address family */
     sai_ip_addr_family_t addr_family;
     union _ip_prefix_addr {
```
Comment (kcudnik) [inline inc/saitypes.h]: can you rebase to master? since unions were extracted to separate instances

Comment (kcudnik) [reply inc/saitypes.h]: hmm, maybe not on v1.2

**Comment by kcudnik** [comment]
closing, this change is for v1.2 branch, if needed just reopen

**Pull request closed without merging**